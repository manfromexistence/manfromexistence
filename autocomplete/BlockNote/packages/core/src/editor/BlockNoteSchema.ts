import {
  defaultBlockSpecs,
  defaultInlineContentSpecs,
  defaultStyleSpecs,
} from "../blocks/defaultBlocks";
import {
  BlockSchema,
  BlockSchemaFromSpecs,
  BlockSpecs,
  InlineContentSchema,
  InlineContentSchemaFromSpecs,
  InlineContentSpecs,
  StyleSchema,
  StyleSchemaFromSpecs,
  StyleSpecs,
  getBlockSchemaFromSpecs,
  getInlineContentSchemaFromSpecs,
  getStyleSchemaFromSpecs,
} from "../schema";
import type {
  BlockNoDefaults,
  PartialBlockNoDefaults,
} from "../schema/blocks/types";
import type { BlockNoteEditor } from "./BlockNoteEditor";

export class BlockNoteSchema<
  BSchema extends BlockSchema,
  ISchema extends InlineContentSchema,
  SSchema extends StyleSchema
> {
  public readonly blockSpecs: BlockSpecs;
  public readonly inlineContentSpecs: InlineContentSpecs;
  public readonly styleSpecs: StyleSpecs;

  public readonly blockSchema: BSchema;
  public readonly inlineContentSchema: ISchema;
  public readonly styleSchema: SSchema;

  // Helper so that you can use typeof schema.BlockNoteEditor
  public readonly BlockNoteEditor: BlockNoteEditor<BSchema, ISchema, SSchema> =
    "only for types" as any;

  public readonly Block: BlockNoDefaults<BSchema, ISchema, SSchema> =
    "only for types" as any;

  public readonly PartialBlock: PartialBlockNoDefaults<
    BSchema,
    ISchema,
    SSchema
  > = "only for types" as any;

  public static create<
    BSpecs extends BlockSpecs = typeof defaultBlockSpecs,
    ISpecs extends InlineContentSpecs = typeof defaultInlineContentSpecs,
    SSpecs extends StyleSpecs = typeof defaultStyleSpecs
  >(options?: {
    /**
     * A list of custom block types that should be available in the editor.
     */
    blockSpecs?: BSpecs;
    /**
     * A list of custom InlineContent types that should be available in the editor.
     */
    inlineContentSpecs?: ISpecs;
    /**
     * A list of custom Styles that should be available in the editor.
     */
    styleSpecs?: SSpecs;
  }) {
    return new BlockNoteSchema<
      BlockSchemaFromSpecs<BSpecs>,
      InlineContentSchemaFromSpecs<ISpecs>,
      StyleSchemaFromSpecs<SSpecs>
    >(options);
    // as BlockNoteSchema<
    // BlockSchemaFromSpecs<BSpecs>,
    // InlineContentSchemaFromSpecs<ISpecs>,
    // StyleSchemaFromSpecs<SSpecs>
    // >;
  }

  constructor(opts?: {
    blockSpecs?: BlockSpecs;
    inlineContentSpecs?: InlineContentSpecs;
    styleSpecs?: StyleSpecs;
  }) {
    this.blockSpecs = opts?.blockSpecs || defaultBlockSpecs;
    this.inlineContentSpecs =
      opts?.inlineContentSpecs || defaultInlineContentSpecs;
    this.styleSpecs = opts?.styleSpecs || defaultStyleSpecs;

    this.blockSchema = getBlockSchemaFromSpecs(this.blockSpecs) as any;
    this.inlineContentSchema = getInlineContentSchemaFromSpecs(
      this.inlineContentSpecs
    ) as any;
    this.styleSchema = getStyleSchemaFromSpecs(this.styleSpecs) as any;
  }
}
