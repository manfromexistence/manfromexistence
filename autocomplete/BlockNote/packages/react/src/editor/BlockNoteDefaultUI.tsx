import { filterSuggestionItems } from "@blocknote/core";
import { FormattingToolbarController } from "../components/FormattingToolbar/FormattingToolbarController";
import { LinkToolbarController } from "../components/LinkToolbar/LinkToolbarController";
import { ImagePanelController } from "../components/ImagePanel/ImagePanelController";
import { SideMenuController } from "../components/SideMenu/SideMenuController";
import { getDefaultReactSlashMenuItems } from "../components/SuggestionMenu/getDefaultReactSlashMenuItems";
import { SuggestionMenuController } from "../components/SuggestionMenu/SuggestionMenuController";
import { TableHandlesController } from "../components/TableHandles/TableHandlesController";
import { useBlockNoteEditor } from "../hooks/useBlockNoteEditor";

export type BlockNoteDefaultUIProps = {
  formattingToolbar?: boolean;
  linkToolbar?: boolean;
  slashMenu?: boolean;
  sideMenu?: boolean;
  imageToolbar?: boolean;
  tableHandles?: boolean;
};

export function BlockNoteDefaultUI(props: BlockNoteDefaultUIProps) {
  const editor = useBlockNoteEditor();

  if (!editor) {
    throw new Error(
      "BlockNoteDefaultUI must be used within a BlockNoteContext.Provider"
    );
  }

  return (
    <>
      {props.formattingToolbar !== false && <FormattingToolbarController />}
      {props.linkToolbar !== false && <LinkToolbarController />}
      {props.slashMenu !== false && (
        <SuggestionMenuController
          getItems={async (query) =>
            filterSuggestionItems(getDefaultReactSlashMenuItems(editor), query)
          }
          // suggestionMenuComponent={MantineSuggestionMenu}
          onItemClick={(item) => {
            item.onItemClick();
          }}
          triggerCharacter="/"
        />
      )}
      {props.sideMenu !== false && <SideMenuController />}
      {editor.imagePanel && props.imageToolbar !== false && (
        <ImagePanelController />
      )}
      {editor.tableHandles && props.tableHandles !== false && (
        <TableHandlesController />
      )}
    </>
  );
}
