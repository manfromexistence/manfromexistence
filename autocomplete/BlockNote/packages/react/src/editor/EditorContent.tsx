import { BlockNoteEditor } from "@blocknote/core";
import { ReactRenderer } from "@tiptap/react";
import { useEffect, useState } from "react";
import { createPortal, flushSync } from "react-dom";

const Portals: React.FC<{ renderers: Record<string, ReactRenderer> }> = ({
  renderers,
}) => {
  return (
    <>
      {Object.entries(renderers).map(([key, renderer]) => {
        return createPortal(renderer.reactElement, renderer.element, key);
      })}
    </>
  );
};

/**
 * Replacement of https://github.com/ueberdosis/tiptap/blob/6676c7e034a46117afdde560a1b25fe75411a21d/packages/react/src/EditorContent.tsx
 * that only takes care of the Portals.
 *
 * Original implementation is messy, and we use a "mount" system in BlockNoteTiptapEditor.tsx that makes this cleaner
 */
export function EditorContent(props: {
  editor: BlockNoteEditor<any, any, any>;
  children: any;
}) {
  const [renderers, setRenderers] = useState<Record<string, ReactRenderer>>({});
  const [singleRenderData, setSingleRenderData] = useState<any>();

  useEffect(() => {
    props.editor._tiptapEditor.contentComponent = {
      /**
       * Used by TipTap
       */
      setRenderer(id: string, renderer: ReactRenderer) {
        setRenderers((renderers) => ({ ...renderers, [id]: renderer }));
      },

      /**
       * Used by TipTap
       */
      removeRenderer(id: string) {
        setRenderers((renderers) => {
          const nextRenderers = { ...renderers };

          delete nextRenderers[id];

          return nextRenderers;
        });
      },

      /**
       * Render a single node to a container within the React context (used by BlockNote renderToDOMSpec)
       */
      renderToElement(node: React.ReactNode, container: HTMLElement) {
        flushSync(() => {
          setSingleRenderData({ node, container });
        });

        // clear after it's been rendered to `container`
        setSingleRenderData(undefined);
      },
    };
    // Without queueMicrotask, custom IC / styles will give a React FlushSync error
    queueMicrotask(() => {
      props.editor._tiptapEditor.createNodeViews();
    });
    return () => {
      props.editor._tiptapEditor.contentComponent = null;
    };
  }, [props.editor._tiptapEditor]);

  return (
    <>
      <Portals renderers={renderers} />
      {singleRenderData &&
        createPortal(singleRenderData.node, singleRenderData.container)}
      {props.children}
    </>
  );
}
