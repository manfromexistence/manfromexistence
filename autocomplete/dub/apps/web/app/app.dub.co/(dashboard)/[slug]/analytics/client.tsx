"use client";

import useDomains from "@/lib/swr/use-domains";
import useWorkspace from "@/lib/swr/use-workspace";
import LayoutLoader from "@/ui/layout/layout-loader";
import LinkNotFound from "@/ui/links/link-not-found";
import WorkspaceExceededClicks from "@/ui/workspaces/workspace-exceeded-clicks";
import { useSearchParams } from "next/navigation";
import { ReactNode } from "react";

export default function AnalyticsClient({ children }: { children: ReactNode }) {
  const { exceededClicks, loading } = useWorkspace();
  if (exceededClicks) {
    return <WorkspaceExceededClicks />;
  }
  const { allDomains, loading: loadingDomains } = useDomains();
  const searchParams = useSearchParams();
  const domain = searchParams?.get("domain");

  if (loading || loadingDomains) {
    return <LayoutLoader />;
  }
  if (
    allDomains?.length === 0 ||
    (domain && !allDomains?.find((d) => d.slug === domain))
  ) {
    return <LinkNotFound />;
  }

  return children;
}
