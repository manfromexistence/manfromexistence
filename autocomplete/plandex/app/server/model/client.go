package model

import (
	"context"
	"log"
	"strings"
	"time"

	"github.com/sashabaranov/go-openai"
)

const OPENAI_STREAM_CHUNK_TIMEOUT = time.Duration(30) * time.Second

func NewClient(apiKey string) *openai.Client {
	config := openai.DefaultConfig(apiKey)
	return openai.NewClientWithConfig(config)
}

func CreateChatCompletionStreamWithRetries(
	client *openai.Client,
	ctx context.Context,
	req openai.ChatCompletionRequest,
) (*openai.ChatCompletionStream, error) {
	return createChatCompletionStream(client, ctx, req, 0)
}

func createChatCompletionStream(
	client *openai.Client,
	ctx context.Context,
	req openai.ChatCompletionRequest,
	numRetry int,
) (*openai.ChatCompletionStream, error) {
	if ctx.Err() != nil {
		return nil, ctx.Err()
	}

	stream, err := client.CreateChatCompletionStream(ctx, req)

	if err != nil {
		log.Printf("Error creating chat completion stream: %v, retry: %d\n", err, numRetry)

		if isNonRetriableErr(err) {
			return nil, err
		}

		// for retriable errors, retry with exponential backoff
		if numRetry < 5 {
			waitBackoff(numRetry)
			return createChatCompletionStream(client, ctx, req, numRetry+1)
		}

		log.Println("Max retries reached - no retry")
		return nil, err
	}

	return stream, nil
}

func CreateChatCompletionWithRetries(
	client *openai.Client,
	ctx context.Context,
	req openai.ChatCompletionRequest,
) (openai.ChatCompletionResponse, error) {
	return createChatCompletion(client, ctx, req, 0)
}

func createChatCompletion(
	client *openai.Client,
	ctx context.Context,
	req openai.ChatCompletionRequest,
	numRetry int,
) (openai.ChatCompletionResponse, error) {

	if ctx.Err() != nil {
		return openai.ChatCompletionResponse{}, ctx.Err()
	}

	resp, err := client.CreateChatCompletion(ctx, req)

	if err != nil {
		log.Printf("Error creating chat completion: %v, retry: %d\n", err, numRetry)

		if isNonRetriableErr(err) {
			return openai.ChatCompletionResponse{}, err
		}

		// for retriable errors, retry with exponential backoff
		if numRetry < 5 {
			waitBackoff(numRetry)
			return createChatCompletion(client, ctx, req, numRetry+1)
		}

		log.Println("Max retries reached - no retry")
		return openai.ChatCompletionResponse{}, err
	}

	return resp, nil
}

func isNonRetriableErr(err error) bool {
	errStr := err.Error()

	// we don't want to retry on the errors below
	if strings.Contains(errStr, "context deadline exceeded") || strings.Contains(errStr, "context canceled") {
		log.Println("Context deadline exceeded or canceled - no retry")
		return true
	}

	if strings.Contains(errStr, "status code: 400") &&
		strings.Contains(errStr, "reduce the length of the messages") {
		log.Println("Token limit exceeded - no retry")
		return true
	}

	if strings.Contains(errStr, "status code: 401") {
		log.Println("Invalid auth or api key - no retry")
		return true
	}

	if strings.Contains(errStr, "status code: 429") && strings.Contains(errStr, "exceeded your current quota") {
		log.Println("Current quota exceeded - no retry")
		return true
	}

	return false
}

func waitBackoff(numRetry int) {
	d := time.Duration(1<<uint(numRetry)) * time.Second
	log.Printf("Retrying in %v\n", d)
	time.Sleep(d)
}
