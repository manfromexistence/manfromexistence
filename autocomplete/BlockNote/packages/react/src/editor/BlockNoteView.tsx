import {
  BlockNoteEditor,
  BlockSchema,
  InlineContentSchema,
  StyleSchema,
  mergeCSSClasses,
} from "@blocknote/core";
import { MantineProvider } from "@mantine/core";

import React, {
  HTMLAttributes,
  ReactNode,
  Ref,
  useCallback,
  useEffect,
  useMemo,
  useState,
} from "react";
import usePrefersColorScheme from "use-prefers-color-scheme";
import { useEditorChange } from "../hooks/useEditorChange";
import { useEditorSelectionChange } from "../hooks/useEditorSelectionChange";
import { mergeRefs } from "../util/mergeRefs";
import { BlockNoteContext, useBlockNoteContext } from "./BlockNoteContext";
import {
  BlockNoteDefaultUI,
  BlockNoteDefaultUIProps,
} from "./BlockNoteDefaultUI";
import {
  Theme,
  applyBlockNoteCSSVariablesFromTheme,
  removeBlockNoteCSSVariables,
} from "./BlockNoteTheme";
import { EditorContent } from "./EditorContent";
import "./styles.css";

const mantineTheme = {
  // Removes button press effect
  activeClassName: "",
};

const emptyFn = () => {
  // noop
};

function BlockNoteViewComponent<
  BSchema extends BlockSchema,
  ISchema extends InlineContentSchema,
  SSchema extends StyleSchema
>(
  props: {
    editor: BlockNoteEditor<BSchema, ISchema, SSchema>;

    theme?:
      | "light"
      | "dark"
      | Theme
      | {
          light: Theme;
          dark: Theme;
        };
    /**
     * Locks the editor from being editable by the user if set to `false`.
     */
    editable?: boolean;
    /**
     * A callback function that runs whenever the text cursor position or selection changes.
     */
    onSelectionChange?: () => void;

    /**
     * A callback function that runs whenever the editor's contents change.
     */
    onChange?: () => void;

    children?: ReactNode;

    ref?: Ref<HTMLDivElement> | undefined; // only here to get types working with the generics. Regular form doesn't work
  } & Omit<
    HTMLAttributes<HTMLDivElement>,
    "onChange" | "onSelectionChange" | "children"
  > &
    BlockNoteDefaultUIProps,
  ref: React.Ref<HTMLDivElement>
) {
  const {
    editor,
    className,
    theme,
    children,
    editable,
    onSelectionChange,
    onChange,
    formattingToolbar,
    linkToolbar,
    slashMenu,
    sideMenu,
    imageToolbar,
    tableHandles,
    ...rest
  } = props;

  const existingContext = useBlockNoteContext();

  const systemColorScheme = usePrefersColorScheme();
  const defaultColorScheme =
    existingContext?.colorSchemePreference || systemColorScheme;

  const [editorColorScheme, setEditorColorScheme] = useState<
    "light" | "dark" | undefined
  >(undefined);

  const containerRef = useCallback(
    (node: HTMLDivElement | null) => {
      if (!node) {
        // todo: clean variables?
        return;
      }

      removeBlockNoteCSSVariables(node);

      if (theme === "light") {
        setEditorColorScheme("light");
        return;
      }

      if (theme === "dark") {
        setEditorColorScheme("dark");
        return;
      }

      if (typeof theme === "object") {
        if ("light" in theme && "dark" in theme) {
          applyBlockNoteCSSVariablesFromTheme(
            theme[defaultColorScheme === "dark" ? "dark" : "light"],
            node
          );
          setEditorColorScheme(
            defaultColorScheme === "dark" ? "dark" : "light"
          );
          return;
        }

        applyBlockNoteCSSVariablesFromTheme(theme, node);
        setEditorColorScheme(undefined);
        return;
      }

      setEditorColorScheme(defaultColorScheme === "dark" ? "dark" : "light");
    },
    [defaultColorScheme, theme]
  );

  useEditorChange(onChange || emptyFn, editor);
  useEditorSelectionChange(onSelectionChange || emptyFn, editor);

  useEffect(() => {
    editor.isEditable = editable !== false;
  }, [editable, editor]);

  const renderChildren = useMemo(() => {
    return (
      <>
        {children}
        <BlockNoteDefaultUI
          formattingToolbar={formattingToolbar}
          linkToolbar={linkToolbar}
          slashMenu={slashMenu}
          sideMenu={sideMenu}
          imageToolbar={imageToolbar}
          tableHandles={tableHandles}
        />
      </>
    );
  }, [
    children,
    formattingToolbar,
    linkToolbar,
    imageToolbar,
    sideMenu,
    slashMenu,
    tableHandles,
  ]);

  const context = useMemo(() => {
    return {
      ...existingContext,
      editor,
    };
  }, [existingContext, editor]);

  const refs = useMemo(() => {
    return mergeRefs([containerRef, editor._tiptapEditor.mount, ref]);
  }, [containerRef, editor._tiptapEditor.mount, ref]);

  return (
    // `cssVariablesSelector` scopes Mantine CSS variables to only the editor,
    // as proposed here:  https://github.com/orgs/mantinedev/discussions/5685
    <MantineProvider theme={mantineTheme} cssVariablesSelector=".bn-container">
      <BlockNoteContext.Provider value={context as any}>
        <EditorContent editor={editor}>
          <div
            className={mergeCSSClasses("bn-container", className || "")}
            data-color-scheme={editorColorScheme}
            {...rest}
            ref={refs}>
            {renderChildren}
          </div>
        </EditorContent>
      </BlockNoteContext.Provider>
    </MantineProvider>
  );
}

export const BlockNoteView = React.forwardRef(
  BlockNoteViewComponent
) as typeof BlockNoteViewComponent; // need hack to get types working with generics
