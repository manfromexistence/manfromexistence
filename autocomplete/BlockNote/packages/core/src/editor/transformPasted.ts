import { Fragment, Slice } from "@tiptap/pm/model";
import { EditorView } from "@tiptap/pm/view";

// helper function to remove a child from a fragment
function removeChild(node: Fragment, n: number) {
  const children: any[] = [];
  node.forEach((child, _, i) => {
    if (i !== n) {
      children.push(child);
    }
  });
  return Fragment.from(children);
}

/**
 * fix for https://github.com/ProseMirror/prosemirror/issues/1430#issuecomment-1822570821
 *
 * Without this fix, pasting two paragraphs would cause the second one to be indented in the other
 * this fix wraps every element in the slice in it's own blockContainer, to prevent Prosemirror from nesting the
 * elements on paste.
 *
 * The exception is when we encounter blockGroups with listitems, because those actually should be nested
 */
export function transformPasted(slice: Slice, view: EditorView) {
  let f = Fragment.from(slice.content);
  for (let i = 0; i < f.childCount; i++) {
    if (f.child(i).type.spec.group === "blockContent") {
      const content = [f.child(i)];

      // when there is a blockGroup with lists, it should be nested in the new blockcontainer
      // (if we remove this if-block, the nesting bug will be fixed, but lists won't be nested correctly)
      if (
        i + 1 < f.childCount &&
        f.child(i + 1).type.spec.group === "blockGroup"
      ) {
        const nestedChild = f
          .child(i + 1)
          .child(0)
          .child(0);

        if (
          nestedChild.type.name === "bulletListItem" ||
          nestedChild.type.name === "numberedListItem"
        ) {
          content.push(f.child(i + 1));
          f = removeChild(f, i + 1);
        }
      }
      const container = view.state.schema.nodes.blockContainer.create(
        undefined,
        content
      );
      f = f.replaceChild(i, container);
    }
  }

  return new Slice(f, slice.openStart, slice.openEnd);
}
