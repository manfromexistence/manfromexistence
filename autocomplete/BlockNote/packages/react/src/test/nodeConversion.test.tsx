import { afterEach, beforeEach, describe, expect, it } from "vitest";

import {
  BlockNoteEditor,
  PartialBlock,
  UniqueID,
  blockToNode,
  nodeToBlock,
  partialBlockToBlockForTesting,
} from "@blocknote/core";
import { flushSync } from "react-dom";
import { Root, createRoot } from "react-dom/client";
import { BlockNoteView } from "../editor/BlockNoteView";
import { customReactBlockSchemaTestCases } from "./testCases/customReactBlocks";
import { customReactInlineContentTestCases } from "./testCases/customReactInlineContent";
import { customReactStylesTestCases } from "./testCases/customReactStyles";

function addIdsToBlock(block: PartialBlock<any, any, any>) {
  if (!block.id) {
    block.id = UniqueID.options.generateID();
  }
  for (const child of block.children || []) {
    addIdsToBlock(child);
  }
}

function validateConversion(
  block: PartialBlock<any, any, any>,
  editor: BlockNoteEditor<any, any, any>
) {
  addIdsToBlock(block);
  const node = blockToNode(
    block,
    editor._tiptapEditor.schema,
    editor.schema.styleSchema
  );

  expect(node).toMatchSnapshot();

  const outputBlock = nodeToBlock(
    node,
    editor.schema.blockSchema,
    editor.schema.inlineContentSchema,
    editor.schema.styleSchema
  );

  const fullOriginalBlock = partialBlockToBlockForTesting(
    editor.schema.blockSchema,
    block
  );

  expect(outputBlock).toStrictEqual(fullOriginalBlock);
}

const testCases = [
  customReactBlockSchemaTestCases,
  customReactStylesTestCases,
  customReactInlineContentTestCases,
];

describe("Test React BlockNote-Prosemirror conversion", () => {
  for (const testCase of testCases) {
    describe("Case: " + testCase.name, () => {
      let editor: BlockNoteEditor<any, any, any>;
      let root: Root;
      const div = document.createElement("div");

      beforeEach(() => {
        editor = testCase.createEditor();

        const el = <BlockNoteView editor={editor} />;
        root = createRoot(div);
        flushSync(() => {
          // eslint-disable-next-line testing-library/no-render-in-setup
          root.render(el);
        });
      });

      afterEach(() => {
        root.unmount();
        editor._tiptapEditor.destroy();
        editor = undefined as any;

        delete (window as Window & { __TEST_OPTIONS?: any }).__TEST_OPTIONS;
      });

      for (const document of testCase.documents) {
        // eslint-disable-next-line no-loop-func
        it("Convert " + document.name + " to/from prosemirror", () => {
          // NOTE: only converts first block
          validateConversion(document.blocks[0], editor);
        });
      }
    });
  }
});
