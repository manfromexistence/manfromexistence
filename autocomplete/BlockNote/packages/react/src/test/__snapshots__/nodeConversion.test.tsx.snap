// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test React BlockNote-Prosemirror conversion > Case: custom react block schema > Convert reactContextParagraph/basic to/from prosemirror 1`] = `
{
  "attrs": {
    "backgroundColor": "default",
    "id": "1",
    "textColor": "default",
  },
  "content": [
    {
      "attrs": {
        "textAlignment": "left",
      },
      "content": [
        {
          "text": "React Context Paragraph",
          "type": "text",
        },
      ],
      "type": "reactContextParagraph",
    },
  ],
  "type": "blockContainer",
}
`;

exports[`Test React BlockNote-Prosemirror conversion > Case: custom react block schema > Convert reactCustomParagraph/basic to/from prosemirror 1`] = `
{
  "attrs": {
    "backgroundColor": "default",
    "id": "1",
    "textColor": "default",
  },
  "content": [
    {
      "attrs": {
        "textAlignment": "left",
      },
      "content": [
        {
          "text": "React Custom Paragraph",
          "type": "text",
        },
      ],
      "type": "reactCustomParagraph",
    },
  ],
  "type": "blockContainer",
}
`;

exports[`Test React BlockNote-Prosemirror conversion > Case: custom react block schema > Convert reactCustomParagraph/nested to/from prosemirror 1`] = `
{
  "attrs": {
    "backgroundColor": "default",
    "id": "1",
    "textColor": "default",
  },
  "content": [
    {
      "attrs": {
        "textAlignment": "left",
      },
      "content": [
        {
          "text": "React Custom Paragraph",
          "type": "text",
        },
      ],
      "type": "reactCustomParagraph",
    },
    {
      "content": [
        {
          "attrs": {
            "backgroundColor": "default",
            "id": "2",
            "textColor": "default",
          },
          "content": [
            {
              "attrs": {
                "textAlignment": "left",
              },
              "content": [
                {
                  "text": "Nested React Custom Paragraph 1",
                  "type": "text",
                },
              ],
              "type": "reactCustomParagraph",
            },
          ],
          "type": "blockContainer",
        },
        {
          "attrs": {
            "backgroundColor": "default",
            "id": "3",
            "textColor": "default",
          },
          "content": [
            {
              "attrs": {
                "textAlignment": "left",
              },
              "content": [
                {
                  "text": "Nested React Custom Paragraph 2",
                  "type": "text",
                },
              ],
              "type": "reactCustomParagraph",
            },
          ],
          "type": "blockContainer",
        },
      ],
      "type": "blockGroup",
    },
  ],
  "type": "blockContainer",
}
`;

exports[`Test React BlockNote-Prosemirror conversion > Case: custom react block schema > Convert reactCustomParagraph/styled to/from prosemirror 1`] = `
{
  "attrs": {
    "backgroundColor": "pink",
    "id": "1",
    "textColor": "orange",
  },
  "content": [
    {
      "attrs": {
        "textAlignment": "center",
      },
      "content": [
        {
          "text": "Plain ",
          "type": "text",
        },
        {
          "marks": [
            {
              "attrs": {
                "stringValue": "red",
              },
              "type": "textColor",
            },
          ],
          "text": "Red Text ",
          "type": "text",
        },
        {
          "marks": [
            {
              "attrs": {
                "stringValue": "blue",
              },
              "type": "backgroundColor",
            },
          ],
          "text": "Blue Background ",
          "type": "text",
        },
        {
          "marks": [
            {
              "attrs": {
                "stringValue": "red",
              },
              "type": "textColor",
            },
            {
              "attrs": {
                "stringValue": "blue",
              },
              "type": "backgroundColor",
            },
          ],
          "text": "Mixed Colors",
          "type": "text",
        },
      ],
      "type": "reactCustomParagraph",
    },
  ],
  "type": "blockContainer",
}
`;

exports[`Test React BlockNote-Prosemirror conversion > Case: custom react block schema > Convert simpleReactCustomParagraph/basic to/from prosemirror 1`] = `
{
  "attrs": {
    "backgroundColor": "default",
    "id": "1",
    "textColor": "default",
  },
  "content": [
    {
      "attrs": {
        "textAlignment": "left",
      },
      "content": [
        {
          "text": "React Custom Paragraph",
          "type": "text",
        },
      ],
      "type": "simpleReactCustomParagraph",
    },
  ],
  "type": "blockContainer",
}
`;

exports[`Test React BlockNote-Prosemirror conversion > Case: custom react block schema > Convert simpleReactCustomParagraph/nested to/from prosemirror 1`] = `
{
  "attrs": {
    "backgroundColor": "default",
    "id": "1",
    "textColor": "default",
  },
  "content": [
    {
      "attrs": {
        "textAlignment": "left",
      },
      "content": [
        {
          "text": "Custom React Paragraph",
          "type": "text",
        },
      ],
      "type": "simpleReactCustomParagraph",
    },
    {
      "content": [
        {
          "attrs": {
            "backgroundColor": "default",
            "id": "2",
            "textColor": "default",
          },
          "content": [
            {
              "attrs": {
                "textAlignment": "left",
              },
              "content": [
                {
                  "text": "Nested React Custom Paragraph 1",
                  "type": "text",
                },
              ],
              "type": "simpleReactCustomParagraph",
            },
          ],
          "type": "blockContainer",
        },
        {
          "attrs": {
            "backgroundColor": "default",
            "id": "3",
            "textColor": "default",
          },
          "content": [
            {
              "attrs": {
                "textAlignment": "left",
              },
              "content": [
                {
                  "text": "Nested React Custom Paragraph 2",
                  "type": "text",
                },
              ],
              "type": "simpleReactCustomParagraph",
            },
          ],
          "type": "blockContainer",
        },
      ],
      "type": "blockGroup",
    },
  ],
  "type": "blockContainer",
}
`;

exports[`Test React BlockNote-Prosemirror conversion > Case: custom react block schema > Convert simpleReactCustomParagraph/styled to/from prosemirror 1`] = `
{
  "attrs": {
    "backgroundColor": "pink",
    "id": "1",
    "textColor": "orange",
  },
  "content": [
    {
      "attrs": {
        "textAlignment": "center",
      },
      "content": [
        {
          "text": "Plain ",
          "type": "text",
        },
        {
          "marks": [
            {
              "attrs": {
                "stringValue": "red",
              },
              "type": "textColor",
            },
          ],
          "text": "Red Text ",
          "type": "text",
        },
        {
          "marks": [
            {
              "attrs": {
                "stringValue": "blue",
              },
              "type": "backgroundColor",
            },
          ],
          "text": "Blue Background ",
          "type": "text",
        },
        {
          "marks": [
            {
              "attrs": {
                "stringValue": "red",
              },
              "type": "textColor",
            },
            {
              "attrs": {
                "stringValue": "blue",
              },
              "type": "backgroundColor",
            },
          ],
          "text": "Mixed Colors",
          "type": "text",
        },
      ],
      "type": "simpleReactCustomParagraph",
    },
  ],
  "type": "blockContainer",
}
`;

exports[`Test React BlockNote-Prosemirror conversion > Case: custom react inline content schema > Convert mention/basic to/from prosemirror 1`] = `
{
  "attrs": {
    "backgroundColor": "default",
    "id": "1",
    "textColor": "default",
  },
  "content": [
    {
      "attrs": {
        "textAlignment": "left",
      },
      "content": [
        {
          "text": "I enjoy working with ",
          "type": "text",
        },
        {
          "attrs": {
            "user": "Matthew",
          },
          "type": "mention",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "blockContainer",
}
`;

exports[`Test React BlockNote-Prosemirror conversion > Case: custom react inline content schema > Convert tag/basic to/from prosemirror 1`] = `
{
  "attrs": {
    "backgroundColor": "default",
    "id": "1",
    "textColor": "default",
  },
  "content": [
    {
      "attrs": {
        "textAlignment": "left",
      },
      "content": [
        {
          "text": "I love ",
          "type": "text",
        },
        {
          "content": [
            {
              "text": "BlockNote",
              "type": "text",
            },
          ],
          "type": "tag",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "blockContainer",
}
`;

exports[`Test React BlockNote-Prosemirror conversion > Case: custom react style schema > Convert fontSize/basic to/from prosemirror 1`] = `
{
  "attrs": {
    "backgroundColor": "default",
    "id": "1",
    "textColor": "default",
  },
  "content": [
    {
      "attrs": {
        "textAlignment": "left",
      },
      "content": [
        {
          "marks": [
            {
              "attrs": {
                "stringValue": "18px",
              },
              "type": "fontSize",
            },
          ],
          "text": "This is text with a custom fontSize",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "blockContainer",
}
`;

exports[`Test React BlockNote-Prosemirror conversion > Case: custom react style schema > Convert small/basic to/from prosemirror 1`] = `
{
  "attrs": {
    "backgroundColor": "default",
    "id": "1",
    "textColor": "default",
  },
  "content": [
    {
      "attrs": {
        "textAlignment": "left",
      },
      "content": [
        {
          "marks": [
            {
              "type": "small",
            },
          ],
          "text": "This is a small text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "blockContainer",
}
`;
